src/opencmiss/zincwidgets/materialeditorwidget.py

```diff
@@ -23,6 +23,116 @@
 from opencmiss.utils.zinc.general import ChangeManager
 
 
+class MaterialModel(QtCore.QAbstractListModel):
+
+    def __init__(self, parent=None):
+        super(MaterialModel, self).__init__(parent)
+        self._material_names = []
+        self._material_module = None
+
+    def setMaterialModule(self, material_module):
+        self._material_module = material_module
+        self._build_material_names_list()
+
+    def getMaterialForIndex(self, index):
+        return self._material_module.findMaterialByName(self._material_names[index.row()])
+
+    def _try_to_remove_material(self, name):
+        can_remove = True
+        mm = self._material_module
+        with ChangeManager(mm):
+            material = mm.findMaterialByName(name)
+            material.setManaged(False)
+            del material
+
+        recovered_material = self._material_module.findMaterialByName(name)
+        if recovered_material.isValid():
+            # I'm still in use.
+            recovered_material.setManaged(True)
+            can_remove = False
+
+        return can_remove
+
+    def _build_material_names_list(self):
+        material_iterator = self._material_module.createMaterialiterator()
+        material = material_iterator.next()
+
+        while material.isValid():
+            self._material_names.append(material.getName())
+            material = material_iterator.next()
+
+    def rowCount(self, parent_index):
+        return len(self._material_names)
+
+    def headerData(self, section, orientation, role):
+        return ['Materials']
+
+    def flags(self, index):
+        if index.isValid():
+            if index.column() == 0:
+                return QtCore.Qt.ItemIsEditable | QtCore.Qt.ItemIsEnabled | QtCore.Qt.ItemIsSelectable
+
+        return QtCore.Qt.NoItemFlags
+
+    def data(self, index, role):
+        if not index.isValid():
+            return None
+
+        if role == QtCore.Qt.DisplayRole:
+            return self._material_names[index.row()]
+
+        return None
+
+    def setData(self, index, value, role):
+        if index.isValid():
+            if role == QtCore.Qt.EditRole:
+                current_name = self._material_names[index.row()]
+                material = self._material_module.findMaterialByName(current_name)
+                result = material.setName(value)
+                if result == ZINC_OK:
+                    self._material_names[index.row()] = value
+                    return True
+
+        return False
+
+    def addData(self):
+        index = self.index(-1, 0)
+        row_count = len(self._material_names)
+        self.beginInsertRows(index, row_count, row_count)
+        name = 'temp'
+        mm = self._material_module
+        with ChangeManager(mm):
+            material = mm.createMaterial()
+            material.setName(name)
+            material.setManaged(True)
+            del material
+        self._material_names.append(name)
```

Comment:
This causes a problem if you press create twice, because the name 'temp' is already in use.
Zinc gives new materials unique names starting with 'temp' followed by a number, I believe starting with the number of elements in the module + 1.
You could accept the name from Zinc in which case you must set `name = material.getName()`.

Alternatively, if you want more control of the name with your own prefix, and starting number here's a possible method to use:
```
    def _getUniqueMaterialName(self, nameStem: str='material'):
        """
        :param nameStem: Starting name, to which a unique integer is added.
        :return: Unique name for a new material in the material module.
        """
        number = 1
        while True:
            name = nameStem + str(number)
            if not self._material_module.findMaterialByName().isValid():
                break
            number += 1
        return name
```
And: `name = self._getUniqueMaterialName()`